TelepresenceBot/mobile/src/main/java/com/novoda/tpbot/support/Observable.java

```diff
@@ -0,0 +1,69 @@
+package com.novoda.tpbot.support;
+
+import com.novoda.notils.exception.DeveloperError;
+
+import java.util.ArrayList;
+
+public abstract class Observable<T> {
+
+    private boolean hasChanged = false;
+    private final ArrayList<Observer<T>> observers;
+
+    protected Observable() {
+        observers = new ArrayList<>();
+    }
+
+    public synchronized Observable<T> attach(Observer<T> observer) {
+        if (observer == null) {
+            throw new DeveloperError("You cannot attach a null observer");
+        }
+
+        if (observers.contains(observer)) {
+            return this;
+        }
+
+        observers.add(observer);
+
+        return this;
+    }
+
+    public synchronized void detach(Observer observer) {
+        observers.remove(observer);
+    }
+
+    protected void notifyOf(T newValue) {
+        Observer<T>[] observersCopy;
+
+        synchronized (this) {
+            if (hasNotChanged()) {
+                return;
+            }
+
+            observersCopy = observers.toArray(new Observer[observers.size()]);
+            clearChanged();
+        }
+
+        for (int i = observersCopy.length - 1; i >= 0; i--) {
+            observersCopy[i].update(newValue);
+        }
+    }
+
+    protected synchronized void detachObservers() {
+        observers.clear();
+    }
+
+    protected synchronized void setChanged() {
+        hasChanged = true;
+    }
+
+    private synchronized void clearChanged() {
```

Comment:
After thinking about this some more I think I would rather keep the `synchronized` it shows intent that this can be used on different threads. Removing it might be misleading to someone in the future.